### ISSUE TITLE
How to test on a custom dataset with ground truth.

### ISSUE THREAD

**Issue by Preksha2** [dvl-tum/SUSHI#31]
I have a question regarding testing on a custom dataset. In your current setup, you seem to be using the training dataset for both training and evaluation (as per your comment in the last line of main.py file: " # Only works if you are testing on train or val data. Will fail in case of a test set"). However, I have a separate testing dataset along with its ground truth annotations for evaluation.

![Image](https://github.com/user-attachments/assets/c682d30b-25c9-4d2c-b756-01a135bc2cbc)

Could you please guide me on how to modify the code so that I can directly test the model on this testing dataset using my ground truth?

Specifically:

I only have ground truth for one dataset (the testing set).

I do not want to train the model; I just want to evaluate it on my dataset.

Where and how should I modify the code to load this dataset and evaluate on it?

**Comment by ocetintas** [comment]
Hi,
What you are referring to is the training/validation setting. I'd recommend you to check the "test" setting under main.py and arrange your dataset in MOTChallenge format. Then you can use motchallenge dataloaders and simply run your evaluation on your dataset of interest.
